### PR TITLE
Read ca files and convert to string before passing to the docker daemon

### PR DESCRIPTION
**Pull request by CameronDiver**
Before this commit, the docker daemon would recieve the filename of the
.pem files, which would be interpreted as the body and would fail. This
commit ensures that the actual body of the pem files are sent to the
daemon.

cc @imrehg 

Change-type: patch
Connects-to: #562
Signed-off-by: Cameron Diver <cameron@resin.io>

### CODE REVIEW

**Comment by imrehg** [comment]
Tried this PR and does connect.

File: lib/utils/docker.coffee [hunk 1]
```diff
@@ -70,36 +70,44 @@ exports.appendOptions = (opts) ->
...
+			# but not all
+			if not (opts.ca? and opts.cert? and opts.key?)
+				throw new Error('You must provide a CA, certificate and key in order to use TLS')
+
+			return Promise.all([fs.readFile(opts.ca), fs.readFile(opts.cert), fs.readFile(opts.key)])
+				.spread (ca, cert, key) ->
+					connectOpts.ca = ca.toString()
+					connectOpts.cert = cert.toString()
+					connectOpts.key = key.toString()
+					return connectOpts
```
Comment (pimterry) [inline lib/utils/docker.coffee]: I think you can slightly simplify this to:

```coffeescript
Promise.props(ca: fs.readFile(opts.ca), cert: fs.readFile(opts.cert), key: fs.readFile(opts.key))
.then (result) ->
  _.mapValues result, value -> value.toString()
```

or even

```coffeescript
Promise.props(_.pick(opts, ['ca', 'cert', 'key']).map(fs.readFile))
.then (result) ->
  _.mapValues(result, value -> value.toString())
```

(although maybe that last step is more terse than it is clear, up to you)

Comment (CameronDiver) [inline lib/utils/docker.coffee]: Huh TIL about `Promise.props`, thanks :)

File: lib/utils/docker.coffee [hunk 2]
```diff
@@ -70,36 +70,44 @@ exports.appendOptions = (opts) ->
...
+		# Now need to check if the user wants to connect over TLS
+		# to the host
+
+		# If any are set...
+		if (opts.ca? or opts.cert? or opts.key?)
+			# but not all
+			if not (opts.ca? and opts.cert? and opts.key?)
+				throw new Error('You must provide a CA, certificate and key in order to use TLS')
+
+			return Promise.props(ca: fs.readFile(opts.ca), cert: fs.readFile(opts.cert), key: fs.readFile(opts.key))
```
Comment (emirotin) [inline lib/utils/docker.coffee]: If you pass the 2nd encoding argument to `readFile` (set it to `utf8`) you shouldn't need `toString`

**Pull request merged**